controllers/install/armadaserver_controller.go

```diff
@@ -0,0 +1,359 @@
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *ArmadaServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	started := time.Now()
+	logger.Info("Reconciling ArmadaServer object")
+
+	logger.Info("Fetching ArmadaServer object from cache")
+	var as installv1alpha1.ArmadaServer
+	if err := r.Client.Get(ctx, req.NamespacedName, &as); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("ArmadaServer not found in cache, ending reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	components, err := generateArmadaServerInstallComponents(&as, r.Scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// TODO remove finalizers (not needed? per Dejan Zele Pejchev)
+
+	deletionTimestamp := as.ObjectMeta.DeletionTimestamp
+	// examine DeletionTimestamp to determine if object is under deletion
+	if deletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !controllerutil.ContainsFinalizer(&as, asFinalizer) {
+			logger.Info("Attaching finalizer to ArmadaServer object", "finalizer", asFinalizer)
+			controllerutil.AddFinalizer(&as, asFinalizer)
+			if err := r.Update(ctx, &as); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		logger.Info("ArmadaServer object is being deleted", "finalizer", asFinalizer)
+		// The object is being deleted
+		if controllerutil.ContainsFinalizer(&as, asFinalizer) {
+			// our finalizer is present, so lets handle any external dependency
+			logger.Info("Running cleanup function for ArmadaServer object", "finalizer", asFinalizer)
+
+			// remove our finalizer from the list and update it.
+			logger.Info("Removing finalizer from ArmadaServer object", "finalizer", asFinalizer)
+			controllerutil.RemoveFinalizer(&as, asFinalizer)
+			if err := r.Update(ctx, &as); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	mutateFn := func() error { return nil }
+
+	logger.Info("Upserting ArmadaServer Deployment object")
+	if components.Deployment != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Deployment, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Ingress object")
+	if components.Ingress != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Ingress, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Ingress_Rest object")
+	if components.Ingress_Rest != nil {
```

Comment:
```suggestion
	if components.IngressRest != nil {
```
Not sure if this is a GoConvention but I don't see snake_case in go code every often.  Everything seems to be CamelCase.

Reply:
I was debating this - was thinking it would map to "ingress-rest' or some such, but you're right, `IngressRest` is better - fixed in commit 29a2d25.